pkg/ebpf/bpf_error.go

```diff
@@ -0,0 +1,223 @@
+
+	BPF_ERR_INIT_CONTEXT
+
+	BPF_ERR_MAP_UPDATE_ELEM
+	BPF_ERR_MAP_DELETE_ELEM
+	BPF_ERR_GET_CURRENT_COMM
+	BPF_ERR_TAIL_CALL
+)
+
+var stringMap = map[BPFErrorType]string{
+	BPF_ERR_UNSPEC: "BPF_ERR_UNSPEC",
+
+	BPF_ERR_INIT_CONTEXT: "BPF_ERR_INIT_CONTEXT",
+
+	BPF_ERR_MAP_UPDATE_ELEM:  "BPF_ERR_MAP_UPDATE_ELEM",
+	BPF_ERR_MAP_DELETE_ELEM:  "BPF_ERR_MAP_DELETE_ELEM",
+	BPF_ERR_GET_CURRENT_COMM: "BPF_ERR_GET_CURRENT_COMM",
+	BPF_ERR_TAIL_CALL:        "BPF_ERR_TAIL_CALL",
+}
+
+var errorMap = map[BPFErrorType]string{
+	BPF_ERR_UNSPEC: "Unspecifed BPF error",
+
+	BPF_ERR_INIT_CONTEXT: "Failed to init context",
+
+	BPF_ERR_MAP_UPDATE_ELEM:  "Failed to add or update a map element",
+	BPF_ERR_MAP_DELETE_ELEM:  "Failed to delete a map element",
+	BPF_ERR_GET_CURRENT_COMM: "Failed to get current command",
+	BPF_ERR_TAIL_CALL:        "Failed to tail call",
+}
+
+func (b BPFErrorType) String() string {
+	if bpfErr, found := stringMap[b]; found {
+		return bpfErr
+	}
+
+	return stringMap[BPF_ERR_UNSPEC]
+}
+
+type BPFError struct {
+	ret  int64
+	id   uint32
+	line uint32
+	file [BPF_MAX_ERR_FILE_LEN]byte
+}
+
+func (b BPFError) Error() string {
+	if bpfErr, found := errorMap[BPFErrorType(b.id)]; found {
+		return bpfErr
+	}
+
+	return errorMap[BPF_ERR_UNSPEC]
+}
+
+func (b BPFError) Return() int64 {
+	return int64(b.ret)
+}
+
+func (b BPFError) ID() uint32 {
+	return uint32(b.id)
+}
+
+func (b BPFError) Type() BPFErrorType {
+	return BPFErrorType(b.id)
+}
+
+func (b BPFError) Line() uint32 {
+	return b.line
+}
+
+func (b BPFError) File() []byte {
+	return b.file[:]
+}
+
+func (b BPFError) FileAsString() string {
+	errFunc := string(b.file[:])
+	errFuncEnd := strings.Index(errFunc, "\x00")
+
+	return errFunc[:errFuncEnd]
+}
+
+func (b BPFError) Size() int {
+	return int(unsafe.Sizeof(b.ret)) +
+		int(unsafe.Sizeof(b.id)) +
+		int(unsafe.Sizeof(b.line)) +
+		int(len(b.file))
+}
+
+func (b *BPFError) Decode(rawBuffer []byte) error {
+	if len(rawBuffer) < b.Size() {
+		return fmt.Errorf("can't decode error raw data - buffer of %d should be of %d bytes", len(rawBuffer), b.Size())
+	}
+
+	b.ret = int64(binary.LittleEndian.Uint64(rawBuffer[0:8]))
+	b.id = binary.LittleEndian.Uint32(rawBuffer[8:12])
+	b.line = binary.LittleEndian.Uint32(rawBuffer[12:16])
+	copy(b.file[:], rawBuffer[16:88])
+
+	return nil
+}
+
+func addUint32(a, b uint32) (uint32, error) {
+	sum := a + b
+	if sum < a {
+		return sum, fmt.Errorf("sum wraped to %v", sum)
+	}
+
+	return sum, nil
+}
+
+// dumpErrorCounts logs error counts at each time interval
+func dumpErrorCounts(t *Tracee) {
+	ticker := time.NewTicker(time.Second * t.config.Output.AggregateBPFErrorsInterval)
+	for range ticker.C {
+		totalCount := uint32(0)
+		var addErr error
+		iter := t.BPFErrorsCountMap.Iterator()
+		for iter.Next() {
+			keyBytes := iter.Key()
+			bpfErr := BPFError{}
+			if err := bpfErr.Decode(keyBytes); err != nil {
+				t.handleError(fmt.Errorf("dumpErrorCounts - %v", err))
```

Comment:
All the errors in this function have the same error message, so it will be hard to distinguish between them if they happen (at least distinguish from which place the rose from).
You should probably add a short description of what failed.